### PR TITLE
EID-2085 Remove eIDAS code and config in MSA

### PR DESCRIPTION
**Pull request by idrop**
See https://govukverify.atlassian.net/browse/EID-2085

## What
Remove eIDAS config and code. Users of the MSA should not have to amend their existing config, so do not remove the `europeanIdentity.enabled` config flag. 

## Why
Now that we have left the EU all redundant eIDAS config and code can be removed. This will make the codebase simpler, and prevent any processing of eIDAS attributes in the MSA.

## Check
- [ ] The `EidasAssertionService` has been removed from the `AttributeQueryService`, which means that the `EidasMatchingDatasetUnmarshaller` and `EidasUnsignedMatchingDatasetUnmarshaller` are never used
- [ ] All `europeanIdentity` config has been removed, apart from `europeanIdentity.enabled`  
- [ ] The `europeanIdentity.enabled` flag in `EuropeanIdentityConfiguration `  is used for the new `MatchingServiceAdapterConfiguration.isUseUniversalMatchingDataset` method

## Note
Release notes have not been updated in this PR. We are waiting to hear from [some RPs about whether its safe to remove all eIDAS config](https://govukverify.atlassian.net/browse/EID-2101).

### CODE REVIEW

File: configuration/compatibility-testing/test-rp-msa-4.2.1-901.yml [hunk 1]
```diff
@@ -81,27 +81,4 @@ encryptionKeys:
 returnStackTraceInErrorResponse: true
 
 europeanIdentity:
-  enabled: true
```
Comment (Wynndow) [inline configuration/compatibility-testing/test-rp-msa-4.2.1-901.yml:84]: This config is used when spinning up an old version of the MSA for the backwards compatibility tests. I think you might have to leave it as is.

File: configuration/compatibility-testing/test-rp-msa-5.0.2.yml [hunk 2]
```diff
@@ -69,23 +69,4 @@ encryptionKeys:
 returnStackTraceInErrorResponse: true
 
 europeanIdentity:
-  enabled: true
```
Comment (Wynndow) [inline configuration/compatibility-testing/test-rp-msa-5.0.2.yml:72]: Same as above for this.

File: prod-config.yml [hunk 3]
```diff
@@ -90,22 +90,6 @@ metadata:
 europeanIdentity:
   enabled: false
 
-  # Details about eIDAS metadata can be configured here
```
Comment (Wynndow) [inline prod-config.yml:93]: ~How does the config class handle values that are supplied in the config file that don't exist as a field? I'm guessing that the services will try to spin up the MSA with their old config file which will include these keys. Are they going to have to remove them?~
I tested this - it doesn't care about extra keys

File: src/integration-test/java/uk/gov/ida/integrationtest/VerifyUserAccountCreationIntegrationTest.java [hunk 4]
```diff
@@ -50,21 +28,4 @@ protected HttpStubRule setUpMatchingService() throws Exception {
         return localMatchingService;
     }
 
-    @Test
```
Comment (Wynndow) [inline src/integration-test/java/uk/gov/ida/integrationtest/VerifyUserAccountCreationIntegrationTest.java:53]: This class now doesn't have any tests - can we just remove it completely?

File: src/main/java/uk/gov/ida/matchingserviceadapter/services/AttributeQueryService.java [hunk 5]
```diff
@@ -45,31 +42,19 @@ public void validate(AttributeQuery attributeQuery) {
...
-            eidasAssertionService.validate(expectedInResponseTo, assertions);
-        }
+        verifyAssertionService.validate(expectedInResponseTo, assertions);
     }
 
     private boolean hasMdsAssertion(List<Assertion> assertions) {
         return assertions.stream()
-                    .map(assertionClassifier::getClassification)
-                    .anyMatch(MDS_ASSERTION::equals);
+                .map(assertionClassifier::getClassification)
```
Comment (Wynndow) [inline src/main/java/uk/gov/ida/matchingserviceadapter/services/AttributeQueryService.java:50]: I don't think `hasMdsAssertion` is used anymore - can probably drop it.

File: src/main/java/uk/gov/ida/matchingserviceadapter/services/EidasAssertionService.java [hunk 6]
```diff
@@ -1,140 +0,0 @@
...
-        this.hubEntityId = hubEntityId;
-        this.matchingDatasetUnmarshaller = matchingDatasetUnmarshaller;
-        this.matchingUnsignedDatasetUnmarshaller = matchingUnsignedDatasetUnmarshaller;
-    }
-
-    @Override
-    void validate(String expectedInResponseTo, List<Assertion> assertions) {
-        for (Assertion assertion : assertions) {
-            if (isCountryAssertion(assertion)) {
-                validateCountryAssertion(assertion, expectedInResponseTo);
```
Comment (Wynndow) [inline src/main/java/uk/gov/ida/matchingserviceadapter/services/EidasAssertionService.java:63]: Good-bye unsigned assertions!

File: configuration/compatibility-testing/test-rp-msa-4.2.1-901.yml (hunk 1 above)
Comment (idrop) [reply configuration/compatibility-testing/test-rp-msa-4.2.1-901.yml:84]: Good spot, will restore!

File: configuration/compatibility-testing/test-rp-msa-5.0.2.yml (hunk 2 above)
Comment (idrop) [reply configuration/compatibility-testing/test-rp-msa-5.0.2.yml:72]: Good spot, will restore!

File: src/integration-test/java/uk/gov/ida/integrationtest/VerifyUserAccountCreationIntegrationTest.java (hunk 4 above)
Comment (idrop) [reply src/integration-test/java/uk/gov/ida/integrationtest/VerifyUserAccountCreationIntegrationTest.java:53]: Oh of course! Will remove

File: src/main/java/uk/gov/ida/matchingserviceadapter/services/AttributeQueryService.java (hunk 5 above)
Comment (idrop) [reply src/main/java/uk/gov/ida/matchingserviceadapter/services/AttributeQueryService.java:50]: Agree, removing

**Review by Wynndow — APPROVED** [review]

**Review by instantiator — APPROVED** [review]

**Pull request merged**